3_javascript/javascript1/week4/script4.js

```diff
@@ -0,0 +1,97 @@
+let aboutMe = [];
+
+function getReply(command) {
+    if (command.includes("Hello my name is")) {
+        const name = command.split("Hello my name is ")[1];
+
+        if (aboutMe.includes(name)) {
+            return "You have already said that!";
+        }
+
+        aboutMe.push(name);
+
+        return `Nice to meet you ${name}`;
+    }
+
+    return "I'm sorry, I don't understand that command.";
+}
+
+//Making a todo list and saving the values in the array aboutMe
+
+function toDoList (activity) {
+if (activity === "fishing" || activity === "singing in the shower") {
+    aboutMe.push(activity);
+return `${activity} added to your todo`;
+}
+}
+
+//Remove fishing from my toDoList
+
+function activityToRemove(activity) {
+    let remainingActivities = aboutMe.filter(todo => todo !== activity);
+    aboutMe = remainingActivities;
+    return `removed ${activity} from your todo list`;
+}
+// what is on my todo
+
+function viewToDo() {
+    const excludedNames = ["Benjamin", "Ben"];
+    const activities = aboutMe.filter(activity => !excludedNames.includes(activity));
+    if (activities.length > 0) {
+        console.log(`Your todo list is: ${activities.join(", ")}`);
+    } else {
+        console.log("Your todo list is empty");
+    }
+}
+
+// What date is it today
+const months = [
+    "January", "February", "March", "April", "May", "June",
+    "July", "August", "September", "October", "November", "December"
+  ];
+const now = new Date();
+const day = now.getDate(); 
+const month = months[now.getMonth()];
+const year = now.getFullYear();
+
+// Should be able to do simple math. fx what is 3 + 3 should respond with 6. Or what is 4 * 12 should respond with 48
+
+function calculate (num1, num2, operation) {
+
+if (operation ==="add") {
+    return num1 + num2;
+
+} else if (operation === "multiply") {
+    return num1 * num2;
+
+} else {
+    return "Operation not supported";
+
+}
+}
+let result = calculate (3,3,"add");
+let result1 = calculate (4,12,"multiply");
+let result2 = calculate (9,3,"Nothing");
+
+// Set a timer for 4 minutes
+setTimeout(myTimeout, 4000);
+
+function myTimeout() {
+    document.getElementById("demo").innerHTML = "I can do magic, woooooo!";
+}
```

Comment:
This should also be in the `getReply` function and will return `Timer done` when the timer is up, and the timer should get triggered by `Set a timer for [number of minutes] minutes` command.

Fx: `getReply(Set a timer for 3 minutes)`